server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/model/MailboxFactory.scala

```diff
@@ -89,11 +79,11 @@ class MailboxFactory @Inject() (subscriptionManager: SubscriptionManager) {
   def create(mailboxMetaData: MailboxMetaData,
              mailboxSession: MailboxSession,
              allMailboxesMetadata: Seq[MailboxMetaData],
-             quotaLoader: QuotaLoader): SMono[Either[Seq[String], Mailbox]] = {
+             quotaLoader: QuotaLoader): SMono[Mailbox] = {
 
     val id: MailboxId = mailboxMetaData.getId
 
-    val name: Either[String, MailboxName] = MailboxName.validate(mailboxMetaData.getPath
+    val name: Either[IllegalArgumentException, MailboxName] = MailboxName.validate(mailboxMetaData.getPath
       .getName
       .split(mailboxSession.getPathDelimiter)
       .last)
```

Comment:
last is not safe to use. will throw a NoSuchElementException If the iterable collection is empty.
you can use `lastOption` and pattern match on it.
Your method will then be something like : 
```
mailboxMetaData.getPath
      .getName
      .split(mailboxSession.getPathDelimiter)
      .lastOption match {
case Some(name) => MailboxName.validate(name)
case None => Left(new IllegalArgumentException("your error message"))
}
    
```